### DISCUSSION TITLE
Patient population

### DISCUSSION THREAD

**Discussion by Sietze-McR** [mobilise-d/mobgap#150] [General]
Which algorithm or pipeline should be used for which patient population?

**Comment by AKuederle** [comment]
Hi @Sietze-McR,

From our testing within Mobilise-D we came up with two slightly different pipelines depending on the patient cohorts. See Fig. 7 here: https://www.nature.com/articles/s41598-024-51766-5
All the algorithms listed there are implemented and can be assembled according to the outlined pipeline.
Note the correction recently published for the algorithm comparison paper: https://link.springer.com/content/pdf/10.1186/s12984-024-01361-6.pdf
The used citation and description within the paper for the GSD_B algorithm was wrong and hence the citation in Fig.7 is also incorrect. A correction will be published soon as well.

In terms of algorithms this means the "normal" `GsdIonescu` instead of the `GsdAdaptiveIonescu` should be used for the P2 pipeline.

These different pipeline structures will also become more clear once #145 is merged. It will implement the pipelines as explicit classes. You can already see the algorithm mapping here:

https://github.com/mobilise-d/mobgap/blob/a8d2c4b3737bcd8a272710d6a127a0da4d5608b5/mobgap/pipeline/_mobilised_pipeline.py#L53

normal_walking -> P1
imparied_walking -> P2
